IOPool/Output/src/RootOutputFile.cc

```diff
@@ -761,10 +756,74 @@ namespace edm {
     }
 
     if(productProvenanceVecPtr != nullptr) productProvenanceVecPtr->assign(provenanceToKeep.begin(), provenanceToKeep.end());
-    treePointers_[branchType]->fillTree();
+    treePointers_[InEvent]->fillTree();
     if(productProvenanceVecPtr != nullptr) productProvenanceVecPtr->clear();
   }
 
+  void RootOutputFile::fillBranches(LuminosityBlockForOutput const& lumi) {
```

Comment:
In order to share the code, would it be better to either make this function templated or to create a base class for all `*ForOutput` classes?